components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/security/APISecurityUtils.java

```diff
@@ -45,6 +45,9 @@ public static void setAuthenticationContext(MessageContext synCtx,
                                                 String contextHeader) {
         synCtx.setProperty(API_AUTH_CONTEXT, authContext);
         synCtx.setProperty(APIConstants.API_KEY_TYPE, authContext.getKeyType());
+        if (authContext.getIssuer() != null) {
```

Comment:
can authContext ever be null?

Reply:
It's "authContext.getIssuer()" issuer is null checked here.

Reply:
What if authContext is null?

Reply:
Checked all occurrences of method calls and params in the current code base; authcontext is not null.